generate.sh

```diff
@@ -0,0 +1,148 @@
+    echo "  to build a reference package.  The generated artifacts are added to the /src/ directory of this repo"
+    echo "  unless a different destination is specified.  If a csv file of packages is specified, multiple"
+    echo "  packages and their dependencies are generated."
+    echo ""
+    echo "  Either --pkg or --pkgCsv must be specified"
+    echo ""
+    echo "options:"
+    echo "  --pkg <packageName,version[,tfm]>  A package and version, no spaces, separated by comma.  If the optional"
+    echo "                                       tfm is specified, it will be used in the project that restores the"
+    echo "                                       package."
+    echo "                                       Example: System.Text.Json,4.7.0"
+    echo "  --pkgCsv <pathToCSV>               A path to a csv file of packages to generate. Format is the same as the --pkg"
+    echo "                                       option above, one per line.  If specified, the --pkg option is ignored."
+    echo "  --dest <pathToDestRepo>            A path to the root of the repo to copy source into."
+    echo ""
+}
+
+packageVersion=
+defaultPathToCSV="$scriptroot/artifacts/targetPackages.csv"
+pathToCSV=
+pathToDestRepo="$scriptroot"
+positional_args=()
+requiredOptionSpecified=false
+while :; do
+    if [ $# -le 0 ]; then
+        break
+    fi
+    lowerI="$(echo "$1" | awk '{print tolower($0)}')"
+    case $lowerI in
+        "-?"|-h|--help)
+            usage
+            exit 0
+            ;;
+        --pkgcsv)
+            if [ ! -f "$2" ]; then
+                echo -e "${RED}ERROR: CSV file not found - $2${NC}"
+                exit 1
+            fi
+            pathToCSV="$(cd "$(dirname "$2")"; pwd)/$(basename "$2")"
+            requiredOptionSpecified=true
+            shift
+            ;;
+        --dest)
+            if [ ! -d "$2" ]; then
+                echo -e "${RED}ERROR: dest not found - $2${NC}"
+                exit 1
+            fi
+            pathToDestRepo="$(cd -P "$2" && pwd)"
+            shift
+            ;;
+        --pkg)
+            packageVersion="$2"
+            requiredOptionSpecified=true
+            shift
+            ;;
+        *)
+            echo -e "${RED}Unrecognized argument '$1'${NC}"
+            usage
+            exit 1
+            ;;
+    esac
+
+    shift
+done
+
+if [ "$requiredOptionSpecified" != "true" ]; then
+    usage
+    exit 1
+fi
+
+if [ "$packageVersion" != "" ]; then
+    mkdir -p "$(dirname "$defaultPathToCSV")"
+    echo "$packageVersion" > "$defaultPathToCSV"
+fi
+
+if [ "$pathToCSV" == "" ]; then
+    pathToCSV="$defaultPathToCSV"
+fi
+
+# Generate restore projects for each target package
+# Packges are restored in their own project to eliminate any conflicts
+# between different versions of the same package.
+INPUT="$pathToCSV"
+OLDIFS=$IFS
+IFS=,
+DEFAULT_TFM=net6.0
```

Comment:
Consider moving this up to the top as it is something that will need to change between branches.

Reply:
Fixed.